src/Microsoft.OData.Core/WriterValidator.cs

```diff
@@ -20,13 +20,32 @@ internal class WriterValidator : IWriterValidator
         /// </summary>
         private readonly ODataMessageWriterSettings settings;
 
+        /// <summary>
+        /// Object pool that stores instances of the IDuplicatePropertyNameChecker.
+        /// </summary>
+        private ObjectPool<IDuplicatePropertyNameChecker> duplicatePropertyNameCheckerObjectPool;
+
         /// <summary>
         /// Creates a WriterValidator instance and binds it to settings.
         /// </summary>
         /// <param name="settings">The ODataMessageWriterSettings instance to bind to.</param>
         internal WriterValidator(ODataMessageWriterSettings settings)
         {
             this.settings = settings;
+            this.duplicatePropertyNameCheckerObjectPool = settings.ThrowOnDuplicatePropertyNames ?
+                new ObjectPool<IDuplicatePropertyNameChecker>(() => new DuplicatePropertyNameChecker()) :
+                new ObjectPool<IDuplicatePropertyNameChecker>(() => new NullDuplicatePropertyNameChecker());

```

Comment:
Since the `NullDuplicatePropertyNameChecker` holds no state, maybe it can be cached or created as a singleton and reused to avoid creating a new instance each time.

Reply:
#Resolved